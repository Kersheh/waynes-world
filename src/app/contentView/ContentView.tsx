import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { RootState } from 'types';
import Home from 'containers/home/Home';
import AddAlbum from 'containers/addAlbum/AddAlbum';
import Library from 'containers/library/Library';
import Error from 'containers/error/Error';
import styles from './ContentView.module.scss';

const ContentView = () => {
  const dispatch = useDispatch();
  const { activeView } = useSelector((state: RootState) => state.app);
  const { message } = useSelector((state: RootState) => state.error);

  // debug; reset store
  if (false) {
    dispatch({ type: 'PURGE', key: 'root', result: () => null })
  }

  return (
    <div className={styles.contentView}>
      {activeView === 'home' && <Home />}
      {activeView === 'addAlbum' && <AddAlbum />}
      {activeView === 'library' && <Library />}

      {message && <Error message={message} />}
    </div>
  );
};

export default ContentView;
