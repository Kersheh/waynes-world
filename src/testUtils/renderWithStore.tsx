import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
