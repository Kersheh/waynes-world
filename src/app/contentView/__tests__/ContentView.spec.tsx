import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import ContentView from '../ContentView';

describe('<ContentView/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(
      <ContentView />
    );

    expect(container).toMatchSnapshot();
  });
});
