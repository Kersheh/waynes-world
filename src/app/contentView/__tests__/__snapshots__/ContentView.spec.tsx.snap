// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentView/> component should render 1`] = `
<div>
  <div
    class="contentView"
  >
    <div
      class="resetBtn"
    />
    <div
      class="home"
    >
      <h2>
        Recently added
      </h2>
      <p>
        No recently added albums found.
      </p>
      <div
        class="albumScroll"
      />
      <div
        class="albumScroll"
      />
    </div>
  </div>
</div>
`;
